docs/preview-environments/drivers/deploy.mdx

```diff
@@ -0,0 +1,30 @@
+---
+id: deploy
+slug: /preview-environments/drivers/deploy
+title: deploy (default)
+---
+
+This preview environments driver can be used to build, push, and deploy an app or addon to Porter. Since this driver does all of this, it is also the default driver. The [`config`](/preview-environments/porter-yaml-reference#resourceconfig) section, when using this driver, supports the following properties:
+
+- `waitForJob` [`Boolean`, optional] - if set to `true`, the driver will wait for the job to complete before proceeding
+- `onlyCreate` [`Boolean`, optional] - if set to `true`, the driver will only create the app or addon but not update it on subsequent runs
+- `build` [`Object`, required] - options for the build step, with the following properties:
+  - `useCache` [`Boolean`, optional] - if set to `true`, the driver will use the image cache (if available) for the build
+  - `method` [`Boolean`, required] - one of `docker`, `pack` ([Paketo Buildpacks](https://paketo.io)) or `registry`
+  - `context` [`String`, optional] - the current build context or working directory to use. Defaults to the current working directory.
+  - `dockerfile` [`String`, required only when `method` is `docker`] - path to the Dockerfile to use, relative to the `context`. Defaults to `./Dockerfile`.
+  - `image` [`String`, required only when `method` is `registry`] - the container image URI to use
+  - `builder` [`String`, required only when `method` is `pack`] - the Paketo [builder](https://paketo.io/docs/concepts/builders) to use
+  - `buildpacks` [`Array of Strings`, optional] - the list of Paketo [buildpacks](https://paketo.io/docs/concepts/buildpacks) to use, when using the `pack` method
+  - `env` [`Map of String keys to String values`, optional] - the environment variables to use for the build
+- `env_groups` [`Array of Objects`, optional] - list of env groups to use to build and deploy the app or addon, supporting the following properties:
+  - `name` [`String`, required] - the name of the env group
+  - `namespace` [`String`, required] - the namespace to use for the env group
```

Comment:
This is a very confusing aspect of preview environments and deserves to be clarified in its own section. We should clarify the difference between the following cases:
- Using the env group driver
- Using the `env_groups` config value (specifying that env groups are getting cloned from one namespace to the other)
- Setting environment variables manually